src/Components/CreateImageWizard/steps/Users/components/UserInfo.tsx

```diff
@@ -27,17 +32,29 @@ import { isUserGroupValid } from '../../../validators';
 
 const UserInfo = () => {
   const dispatch = useAppDispatch();
-  const index = 0;
-  const userNameSelector = selectUserNameByIndex(index);
-  const userName = useAppSelector(userNameSelector);
-  const userPasswordSelector = selectUserPasswordByIndex(index);
-  const userPassword = useAppSelector(userPasswordSelector);
-  const userSshKeySelector = selectUserSshKeyByIndex(index);
-  const userSshKey = useAppSelector(userSshKeySelector);
-  const userIsAdministratorSelector = selectUserAdministrator(index);
-  const userIsAdministrator = useAppSelector(userIsAdministratorSelector);
-  const userGroupsSelector = selectUserGroupsByIndex(index);
-  const userGroups = useAppSelector(userGroupsSelector);
+  const users = useAppSelector(selectUsers);
+  const stepValidation = useUsersValidation();
+
+  const [index, setIndex] = useState(0);
+  const [activeTabKey, setActiveTabKey] = useState(0);
+  const [tabIndex, setTabIndex] = useState(0);
+  const [showRemoveUserModal, setShowRemoveUserModal] = useState(false);
+
+  const onSelect = (event: React.MouseEvent, tabIndex: number) => {
+    setActiveTabKey(tabIndex);
+    setIndex(tabIndex);
+  };
+
+  const onAdd = () => {
+    setActiveTabKey(users.length);
+    setIndex(index + 1);
+    dispatch(addUser());
+  };
+
+  const onClose = (_event: React.MouseEvent, tabIndex: number) => {
+    setShowRemoveUserModal(true);
```

Comment:
I was wondering. In case no values are on input - name, password, ssh key, and groups are empty - do we want to remove them without the dialog? Just a thought for improvement.

Reply:
Makes sense, updated :+1: